Assignment4/dloriente-u12a065/task06.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+"""Task06.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1VfAfMYhKMGDxIxP8CmSp8MWx58TbByhM
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib 
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example5.rdf", format="xml")
+
+"""Create a new class named Researcher"""
+
+ns = Namespace("http://somewhere#")
+g.add((ns.Researcher, RDF.type, RDFS.Class))
+for s, p, o in sorted(g):
+  print(s,p,o)
+
+"""**TASK 6.1: Create a new class named "University"**
+
+"""
+
+# TO DO
+g.add((ns.University, RDF.type, RDFS.Class))
+
+# Visualize the results: 
+for s, p, o in sorted(g):
+  print(s,p,o)
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+# TO DO
+g.add((ns.Researcher, RDFS.subClassOf, ns.Person))
+
+# Visualize the results: 
+for s, p, o in sorted(g):
+  print(s,p,o)
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+# TO DO
+g.add((ns.JaneSmith, RDF.type, ns.Researcher))
+# FullName "Jane Smith" is added in Task 6.4
+
+# Visualize the results: 
+for s, p, o in sorted(g):
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the fullName, given and family names**"""
+
+# TO DO
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+from rdflib.namespace import XSD
+from rdflib.plugins.sparql import prepareQuery
+
+g.add((ns.JaneSmith, vcard.FN, Literal("Jane Smith", datatype=XSD.string)))
+g.add((ns.JaneSmith, vcard.Given, Literal("Jane", datatype=XSD.string)))
+g.add((ns.JaneSmith, vcard.Family, Literal("Smith", datatype=XSD.string)))
+
+# Visualize the results: all properties of JaneSmith
+q64 = prepareQuery('''
+  SELECT ?Pred ?Obj 
+  WHERE { 
+    ns:JaneSmith ?Pred ?Obj.
+  }
+   ''',
+  initNs = { "ns" : "http://somewhere#"}
+)
+for r in g.query(q64):
+  print(r.Pred, r.Obj)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JohnSmith, vcard.work, ns.UPM))
```

Comment:
vcard does not define the property "work". Define your own

Reply:
Done